packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -308,49 +305,63 @@ contract VotingV2 is
 
         // Traverse all requests from the last considered request. For each request see if the voter voted correctly or
         // not. Based on the outcome, attribute the associated slash to the voter.
-        uint256 priceRequestIdsLength = priceRequestIds.length;
-        for (uint256 i = voterStake.lastRequestIndexConsidered; i < priceRequestIdsLength; i = unsafe_inc(i)) {
-            if (deletedRequestJumpMapping[i] != 0) i = deletedRequestJumpMapping[i] + 1;
-            PriceRequest storage priceRequest = priceRequests[priceRequestIds[i].requestId];
-            VoteInstance storage voteInstance = priceRequest.voteInstances[priceRequest.lastVotingRound];
-            uint256 roundId = priceRequestIds[i].roundId;
-
-            // Cant slash this or any subsequent requests if the request is not settled. TODO: this has implications for
-            // rolled votes and should be considered closely.
+        for (
+            uint256 requestIndex = voterStake.lastRequestIndexConsidered;
+            requestIndex < priceRequestIds.length;
+            requestIndex = unsafe_inc(requestIndex)
+        ) {
+            if (deletedRequests[requestIndex] != 0) requestIndex = deletedRequests[requestIndex] + 1;
```

Comment:
I'm not strongly opinionated on this but I have some concerns that I would like to share and get your feedback 
I think it can be risky to change the requestIndex loop tracker outside of the "for(;;)" other than by incrementing or decrementing it:
1. Makes it hard to reason about
2. Makes it more vulnerable to unexpected bugs because of 1.

I wonder if we could get the same result with two different loops, one for the priceRequestIds and one for the deletedRequests. There might be subtleties or things that wouldn't work. What do you think?

Reply:
OK, I understand the reasoning. Sounds good.